application/src/main/java/org/opentripplanner/apis/support/graphql/injectdoc/CustomDocumentation.java

```diff
@@ -0,0 +1,171 @@
+   * </ol>
+   * @param typeName Use {@code TYPE_NAME} or {@code TYPE_NAME.FIELD_NAME} as key.
+   */
+  public Optional<String> typeDescription(String typeName, @Nullable String originalDoc) {
+    return text(typeName, DESCRIPTION_SUFFIX, originalDoc);
+  }
+
+  /**
+   * Same as {@link #typeDescription(String, String)} except the given {@code typeName} and
+   * {@code fieldName} is used as the key.
+   * <pre>
+   * key := typeName + "." fieldNAme
+   * </pre>
+   */
+  public Optional<String> fieldDescription(
+    String typeName,
+    String fieldName,
+    @Nullable String originalDoc
+  ) {
+    return text(key(typeName, fieldName), DESCRIPTION_SUFFIX, originalDoc);
+  }
+
+  /**
+   * Get <em>deprecated reason</em> for a field (types cannot be deprecated). The key
+   * ({@code key = typeName + '.' + fieldName} is used to retrieve the reason from the properties
+   * file. The deprecated documentation text is resolved by:
+   * <ol>
+   *   <li>
+   *     first looking up the given {@code key} + {@code ".deprecated"}. If a value is found, then
+   *     the value is returned.
+   *   <li>
+   *     then {@code key} + {@code ".deprecated.append"} is used. If a value is found the
+   *     {@code originalDoc} + {@code text} is returned.
+   *   </li>
+   * </ol>
+   * Any {@code null} values are excluded from the result and if both the input {@code originalDoc}
+   * and the resolved value is {@code null}, then {@code empty} is returned.
+   */
+  public Optional<String> fieldDeprecatedReason(
+    String typeName,
+    String fieldName,
+    @Nullable String originalDoc
+  ) {
+    return text(key(typeName, fieldName), DEPRECATED_SUFFIX, originalDoc);
+  }
+
+  /* private methods */
+
+  /**
+   * Create a key from the given {@code typeName} and {@code fieldName}
+   */
+  private static String key(String typeName, String fieldName) {
+    return typeName + "." + fieldName;
+  }
+
+  private Optional<String> text(String key, String suffix, @Nullable String originalText) {
+    final String k = key + suffix;
+    return text(k).or(() -> appendText(k, originalText));
+  }
+
+  private Optional<String> text(String key) {
+    return Optional.ofNullable(textMap.get(key));
+  }
+
+  private Optional<String> appendText(String key, @Nullable String originalText) {
+    String value = textMap.get(key + APPEND_SUFFIX);
+    if (value == null) {
+      return Optional.empty();
+    }
+    return originalText == null ? Optional.of(value) : Optional.of(originalText + "\n\n" + value);
+  }
+
+  /* private methods */
+
+  private static Map<String, String> loadCustomDocumentationFromPropertiesFile(
+    ApiDocumentationProfile profile
+  ) {
+    try {
+      final String resource = resourceName(profile);
+      var input = ClassLoader.getSystemResourceAsStream(resource);
+      if (input == null) {
+        throw new OtpAppException("Resource not found: %s", resource);
+      }
+      var props = new Properties();
+      props.load(input);
```

Comment:
This method assumes that the input stream is ISO-8869-1 since properties files used to be hardcoded to that encoding. 

You need to wrap the input stream in a reader like this:

```
new InputStreamReader(input, "UTF-8")
```